src/components/video.js

```diff
@@ -16,46 +17,81 @@ const sampleData = {
   spotify: 'https://open.spotify.com/track/4uLU6hMCjMI75M1A2tKUQC',
   likes: 27,
   genres: ['soul', 'pop'],
-  queue: [
-    'zReP_EYZGEw',
-    'dQw4w9WgXcQ',
-    'PJWemSzExXs',
-    'k4V3Mo61fJM',
-    'e-ORhEE9VVg',
-    '2hlT8CqZ2pA',
-    'U0V1y2p1sgs',
-  ],
+  queue: [],
 };
 
 export default function Video(props) {
-  let preQueue;
+  let preQueue = useSelector((state) => state.queue);
   if (props.queue) {
     preQueue = props.queue;
-  } else {
-    preQueue = sampleData.queue;
   }
   let [queue, changeQueue] = useState([
     props.id || sampleData.yid,
     ...preQueue,
   ]);
 
+  let [currentIndex, changeIndex] = useState(0);
+
   useEffect(() => {
     if (document !== undefined) {
       document.body.scrollTop = 0; // For Safari
       document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
     }
   }, []);
 
+  let changeURLid = (id) => {
+    if (window !== undefined) {
+      if (window.history.pushState) {
+        var newurl =
+          window.location.protocol +
+          '//' +
+          window.location.host +
+          window.location.pathname +
+          '?=' +
+          id;
+        window.history.pushState({ path: newurl }, '', newurl);
+      }
+    }
+  };
+
+  let scrollCurrentVideo = (top) => {
+    if (document !== undefined) {
+      let el = document.querySelector('.current.queueCard');
+      let Q = document.querySelector('.queue');
+      if (el !== null && Q !== null) {
+        const elementRect = el.getBoundingClientRect();
+        const queueRect = Q.getBoundingClientRect();
+        window.scrollTo(0, elementRect.top - queueRect.y + top);
+      }
+    }
+  };
+
+  let playNextVideo = () => {
```

Comment:
And I've tried understanding but have failed thus far, but have you noticed how if the song ends and the next video automatically starts playing, the `current` card scrolls past the view? If you understand, please explain it to me too.

Reply:
Can you test the one with new changes and check if it has been solved? It works fine now.